### PR TITLE
Task yc/j70 add react select to our select component

### PR DESCRIPTION
**Pull request by EugeniusGR**

### CODE REVIEW

**Comment by EugeniusGR** [comment]
[Add react-select to our select component](https://trello.com/c/oX9Y9ilU/70-add-react-select-to-our-select-component)

File: frontend/src/components/common/select/select.tsx [hunk 1]
```diff
@@ -1,41 +1,35 @@
...
 
-const Select: React.FC<Props> = ({
-  options,
-  label,
-  name,
-  control,
-  errors,
-}) => {
-  const { field } = useController({ name, control });
+import SelectReact from 'react-select';
```
Comment (yevhenprybytok) [inline frontend/src/components/common/select/select.tsx]: Import order

**Review by yevhenprybytok — CHANGES_REQUESTED** [review]

File: frontend/package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "react-redux-toastr": "7.6.5",
     "react-router-dom": "5.2.0",
     "react-scripts": "4.0.3",
+    "react-select": "^4.3.1",
```
Comment (ghost) [inline frontend/package.json]: check **B17** criteria

File: frontend/package.json [hunk 3]
```diff
@@ -35,6 +36,7 @@
     "@types/react-redux": "7.1.18",
     "@types/react-redux-toastr": "7.6.1",
     "@types/react-router-dom": "5.1.8",
+    "@types/react-select": "^4.0.17",
```
Comment (ghost) [inline frontend/package.json]: check **B17** criteria

File: frontend/src/components/common/select/select.tsx [hunk 4]
```diff
@@ -1,41 +1,33 @@
...
-  Path,
-  FieldValues,
-} from 'react-hook-form';
-
+import { Control, FieldErrors, Path, FieldValues } from 'react-hook-form';
 import styles from './styles.module.scss';
 
 type Props = {
-  options: string[],
+  options: string[];
```
Comment (ghost) [inline frontend/src/components/common/select/select.tsx]: now we need to create

```tsx
type Option = {
  label: string;
  value: string;
};
```
and use them for props for the select

File: frontend/src/components/common/select/select.tsx [hunk 5]
```diff
@@ -1,41 +1,33 @@
...
+    return { value: index, label: item };
+  });
 
   return (
     <label className={styles.inputWrapper}>
       <span className={styles.label}>{label}</span>
-      <select
-        {...field}
+      <SelectReact
+        defaultValue={optionsItems[0]}
```
Comment (ghost) [inline frontend/src/components/common/select/select.tsx]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/select/select.tsx [hunk 6]
```diff
@@ -1,41 +1,37 @@
...
 
-const Select: React.FC<Props> = ({
-  options,
-  label,
-  name,
-  control,
-  errors,
-}) => {
-  const { field } = useController({ name, control });
+type Options = {
```
Comment (ghost) [inline frontend/src/components/common/select/select.tsx]:
```suggestion
type Option = {
```
move it to `frontend/common/types/ui`

File: frontend/src/components/common/select/select.tsx [hunk 7]
```diff
@@ -1,41 +1,37 @@
...
-  Path,
-  FieldValues,
-} from 'react-hook-form';
-
+import { Control, FieldErrors, Path, FieldValues } from 'react-hook-form';
 import styles from './styles.module.scss';
 
 type Props = {
-  options: string[],
+  options: string[];
```
Comment (ghost) [inline frontend/src/components/common/select/select.tsx]:
```suggestion
  options: Option[];
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by Denidzio — DISMISSED** [review]

File: frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx [hunk 8]
```diff
@@ -11,18 +11,20 @@ type Props = {
...
+  });
 
   const { dataStatus } = useAppSelector(({ episode }) => ({
     dataStatus: episode.dataStatus,
   }));
 
   const isFormDisable = dataStatus === DataStatus.PENDING;
 
-  const episodeTypes = Object.values(EpisodeType);
+  const selectOptions:Option[] = Object.values(EpisodeType).map((item)=>{
```
Comment (ghost) [inline frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx]: create a helper for it
`getOptions(string[]) {}`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**